x/transfermiddleware/ibc_middleware.go

```diff
@@ -127,6 +127,41 @@ func (im IBCMiddleware) OnTimeoutPacket(ctx sdk.Context, packet channeltypes.Pac
 		return err
 	}
 
+	denom := data.Denom
+	coin := im.keeper.IbcTransfermiddleware.GetCoin(ctx, packet.SourceChannel, denom)
+	сhannelFeeAddress := im.keeper.IbcTransfermiddleware.GetChannelFeeAddress(ctx, packet.SourceChannel)
+	if coin != nil {
+		amount := data.Amount
+		transferAmount, ok := sdk.NewIntFromString(amount)
+		if !ok {
+			return errors.Wrapf(transfertypes.ErrInvalidAmount, "unable to parse transfer amount: %s", amount)
+		}
+
+		// calculate the percentage charge
+		/*
+			coin.Percentage is 100 that means that we charge 1/100 of the transfer amount
```

Comment:
do we charge 1%? Where does this number come from?

Reply:
it is example.
of explanation.
we charge depends on config.
I described how it calculates.
To find a percentage fee from a new amount